### PR TITLE
Add ReactMarkerConstants.CONTENT_APPEARED support on Android

### PR DESCRIPTION
**Pull request by Kudo**
## Summary:

Add the `ReactMarkerConstants.CONTENT_APPEARED` support on Android in bridgeless mode. This is an important marker for TTI measurement.

## Changelog:

[ANDROID] [ADDED] - Add the `ReactMarkerConstants.CONTENT_APPEARED` support on Android in bridgeless mode. 

## Test Plan:

adding this on RNTesterActivity to see if the log is executed

```kotlin
ReactMarker.addListener { name, tag, instanceKey ->
  if (name == ReactMarkerConstants.CONTENT_APPEARED) {
    Log.i("XXX", "XXX")
  }
}
```

### CODE REVIEW

**Comment by Kudo** [comment]
appreciated if @cortinico could take a look for this pr.

**Comment by cortinico** [comment]
cc @rubennorte

**Comment by Kudo** [comment]
rebase main to fix conflict

**Comment by Kudo** [comment]
much appreciated if @rubennorte could help to take a look for this pr

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java [hunk 1]
```diff
@@ -434,10 +434,8 @@ public void run() {
 
     if (mShouldLogContentAppeared) {
       mShouldLogContentAppeared = false;
-
-      if (mJSModuleName != null) {
```
Comment (rubennorte) [inline packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java:438]: What's the motivation for this change? It looks safer to check if it's defined than to assert if it isn't. I'd rather miss this marker than make the app crash.

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java [hunk 2]
```diff
@@ -434,10 +434,8 @@ public void run() {
...
-        ReactMarker.logMarker(ReactMarkerConstants.CONTENT_APPEARED, mJSModuleName, mRootViewTag);
-      }
+      String jsModuleName = getJSModuleName();
```
[the remark this answers is not in the document]
Comment (rubennorte) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java:437]: > could help us catching the issue as early as possible

I agree with this when adding a new API, but modifying an API to introduce this, if the previous behavior wasn't crashing, is not ideal.

Given that we already have some callers in the critical path (e.g.: `startSurface`) that were using the getter with the assert, I think it might be safe to continue using it, even for this use case.

[the remark this answers is not in the document]
Comment (rubennorte) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java]: Sorry I think you misunderstood. I meant that you can keep the implementation you had replacing all usages with `getJSModuleName` with the assert because we were already calling that when loading a new surface. If there were cases where it was defined, that should've thrown an error in those cases too. So replacing all with `getJSModuleName` and removing the null checks should be fine.

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactRoot.java [hunk 3]
```diff
@@ -28,6 +28,7 @@ public interface ReactRoot {
   @Nullable
   Bundle getAppProperties();
 
+  @Nullable
```
Comment (rubennorte) [inline packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactRoot.java:31]: We can remove this `@Nullable` annotation and all the new asserts on this diff

Comment (Kudo) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactRoot.java]: updated. does the diff look correct now?

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java [hunk 4]
```diff
@@ -434,9 +434,9 @@ public void run() {
 
     if (mShouldLogContentAppeared) {
       mShouldLogContentAppeared = false;
-
-      if (mJSModuleName != null) {
-        ReactMarker.logMarker(ReactMarkerConstants.CONTENT_APPEARED, mJSModuleName, mRootViewTag);
+      String jsModuleName = getJSModuleName();
+      if (jsModuleName != null) {
```
Comment (rubennorte) [inline packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java:438]: This is never going to be null now :)

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java [hunk 5]
```diff
@@ -632,7 +632,7 @@ public void setEventListener(@Nullable ReactRootViewEventListener eventListener)
 
   @Override
   public String getJSModuleName() {
-    return Assertions.assertNotNull(mJSModuleName);
+    return mJSModuleName;
```
Comment (rubennorte) [inline packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java:635]: I thought we were adding back the assertion. Can you please update this?

Comment (Kudo) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java]: haha it looks like the original version is more correct. sorry i totally misunderstood what you said before.
updated and please let me know if there are any other feedback.

**Review by rubennorte — APPROVED** [review]
Thanks for adding support for this in the new architecture! I'm going to import this at Meta and test it internally before merging.

**Comment by Kudo** [comment]
oh no i'm sorry if that cause some breaks in meta internally. let me know if there's anything i can help with.

**Comment by Kudo** [comment]
> > oh no i'm sorry if that cause some breaks in meta internally. let me know if there's anything i can help with.
> 
> Hey, don't worry. It's just one CI job that didn't run correctly in the original PR. I'll re-commit it with the fix today/tomorrow.

thanks! feel free to let me know if there's anything i can help.
